docs/semgrep-code/remove-duplicates.md

```diff
@@ -3,29 +3,111 @@ slug: remove-duplicates
+
+This information is combined and then hashed. At this point, Semgrep appends the **index**, a value generated by determining the number of times the rule involved matched code in the file. Note that the index is appended to the hash, not combined with the other finding information before hashing. This is done to preserve information on how findings are related. For example, `finding0` with `match_based_id = 123_0` and `finding1` with `match_based_id = 123_1` indicate that both were generated from the same rule matching the same code pattern in the same file.
+
+Because Semgrep uses the rule pattern instead of the literal code syntax to generate `match_based_id`, code changes that don't impact the code pattern matched also don't hinder Semgrep's ability to recognize that the finding isn't a duplicate of an existing finding.
+
+![Semgrep AppSec Platform groups together findings on different branches](/img/matched-findings.png)
+_**Figure**. Semgrep AppSec Platform groups together the same finding identified as present on multiple branches._
+
+For example, if the original file scanned is:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` is:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+If the following change is made to the original file:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+c_1 = 5
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` does not change:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+This means that the `match_based_id` itself doesn't change, allowing Semgrep to identify that the two findings are the same, allowing deduplication. Furthermore, this process enables Semgrep to ignore lines that do not impact code function. For example, adding comments to a code file would create a new `syntactic_id`, but not a new `match_based_id`.
+
+### `syntactic_id`
+
+Semgrep generates the `syntactic_id` for a finding using the following information:
+
+- The file path
+- The name of the rule that generated the finding
+- The code syntax, or the literal piece of code that matched the rule
+- The index, a value generated by determining the number of times the rule involved matched code in the file
+
+This information is combined and then hashed before being stored for privacy.
+
+Using the `syntactic_id`, Semgrep can determine if a given finding in a file is 
+the same as a finding identified during a different scan, even if the code snippet that the rule matched had been moved to a different location in the file. This allows Semgrep to avoid generating a new finding and deduplicating its records accordingly, even across multiple branches associated with the project. It also means that Semgrep can cross-correlate findings, so a finding that has been triaged in one branch will be flagged as triaged if it's identified in another branch.
```

Comment:
This is actually still not quite right I think. We use the match-based ID for triage checks (I know because I spend a lot of time looking at the configs and results exchanged by the platform and the CLI). @pangngan what do you think about how much we should share about syntactic_id? I know there are cases where we use it in the CLI but I've never understood entirely why/how. Maybe it's enough to say that it's an older ID and is used as a fallback or for internal debugging?

Reply:
You're right -- I think this makes more sense for match ID so I'll change it out and move it. I think the note on why we have syntactic_id should be sufficient